src/metrics.go

```
@@ -60,23 +67,38 @@ func populateMetrics(i *integration.Integration, client Client, env string) {
 	}
 }
 
-func getLocalNodeID(client Client) (nodeId string, err error) {
+// getClusterNeeded determines if cluster and common metrics should be scraped
+func getClusterNeeded(client Client, nodeID string) (needed bool) {
+	if args.MasterOnly || args.LocalOnly {
+		masterID, err := getMasterNodeID(client)
+		if err != nil {
+			log.Error("There was an error gathering the elected master node ID: %v", err)
```

We should not hide errors, The signature likely should be 
`func getClusterNeeded(client Client, nodeID string) (bool, Error) {`

But then it is again difficult to read 🤔

---

updated function name